### PR TITLE
Add refresh

### PR DESCRIPTION
**Pull request by digit01Wave**
#969 

This PR adds the following
- Adds shortcut to Puzzle help thread from "Jump to..."
image
- Make sure you can't edit or post new messages after answers have been released
- Introduce the ability to refresh the threads page for GCs
image

### CODE REVIEW

File: ServerCore/Pages/Threads/PuzzleThread.cshtml.cs [hunk 1]
```diff
@@ -352,5 +370,38 @@ private async Task SendEmailNotifications(Message newMessage, Puzzle puzzle)
...
+
+        private ValidationResult IsMessageTextValid(string messageText)
+        {
+            if (string.IsNullOrWhiteSpace(messageText))
+            {
+                return ValidationResult.CreateFailure("Your message cannot be empty.");
+            }
+            else if (messageText.Length > MessageCharacterLimit)
+            {
+                return ValidationResult.CreateFailure("Your message should not be longer than 3000 characters.");
```
[the remark this answers is not in the document]
Comment (morganbr) [reply ServerCore/Pages/Threads/PuzzleThread.cshtml.cs:382]: If there isn't already one, you can set a limit on the class in Data, which will get enforced by client and server

File: ServerCore/Pages/Threads/PuzzleThread.cshtml.cs [hunk 2]
```diff
@@ -155,13 +155,15 @@ public async Task<IActionResult> OnGetAsync(int? puzzleId, int? teamId, int? pla
 
         public async Task<IActionResult> OnPostAsync()
         {
-            if (string.IsNullOrWhiteSpace(NewMessage.Text))
+            if (Event.AreAnswersAvailableNow)
```
Comment (morganbr) [inline ServerCore/Pages/Threads/PuzzleThread.cshtml.cs:158]: Do we actually need this check? I'm not sure asking for help late is bad per se, it just might not get answered

File: ServerCore/Pages/Threads/PuzzleThread.cshtml.cs [hunk 3]
```diff
@@ -352,5 +370,38 @@ private async Task SendEmailNotifications(Message newMessage, Puzzle puzzle)
                 MailHelper.Singleton.SendPlaintextWithoutBcc(recipients.Select(r => r.Email), emailTitle, emailContent);
             }
         }
+
+        private ValidationResult IsMessageTextValid(string messageText)
+        {
+            if (string.IsNullOrWhiteSpace(messageText))
```
Comment (morganbr) [inline ServerCore/Pages/Threads/PuzzleThread.cshtml.cs:376]: Putting RequiredAttribute on the property would prevent this and add client-side validation

File: ServerCore/Pages/Threads/PuzzleThread.cshtml.cs (hunk 2 above)
Comment (digit01Wave) [reply ServerCore/Pages/Threads/PuzzleThread.cshtml.cs:158]: I just didn't want people continuing to add stuff to the server. 
Also, if the answers are available, would you need help?

**Review by morganbr — APPROVED** [review]

File: ServerCore/Pages/Threads/PuzzleThread.cshtml.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (digit01Wave) [reply ServerCore/Pages/Threads/PuzzleThread.cshtml.cs:382]: the message remains. For example, this is what it looks like when I change the limit to 10
image

I have added maxlength to the client side, but someone can just adjust the html to get around it

**Pull request merged**